capstone/src/main/webapp/scripts/poll.js

```diff
@@ -74,6 +80,9 @@ function getChallenge() {
     .then(() => checkWeek(dueDateMillis));
```

Comment:
here you're calling `checkWeek(dueDateMillis)` regardless of `if (challengeData)` was true, meaning `dueDateMillis` may still be 0. Is this intended?

If not, consider moving checkWeek into `if (challengeData)`